powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/client/activation/PowerAuthClientActivation.java

```diff
@@ -45,8 +45,16 @@
 public class PowerAuthClientActivation {
 
     private final SignatureUtils signatureUtils = new SignatureUtils();
-    private final KeyGenerator keyGenerator = new KeyGenerator();
     private final KeyConvertor keyConvertor = new KeyConvertor();
+    private final KeyGenerator keyGenerator;
+
+    /**
+     * PowerAuth client activation constructor.
+     * @throws CryptoProviderException In case key cryptography provider is incorrectly initialized.
+     */
+    public PowerAuthClientActivation() throws CryptoProviderException {
+        keyGenerator = new KeyGenerator();
```

Comment:
It seems a bit weird we only initialize `keyGenerator` in the constructor - is there a way to use the inline field initialization as we do for `keyConveror`? Maybe making a `CryptoProviderException` runtime?

Reply:
I do not think this is a good idea. The checked exception tells the client code that this issue can happen and we know it can happen quite easily when BC is not set up correctly. It is better to let the client handle it including proper logging (and other cleanup if needed) than interrupting the processing unexpectedly.